plugins/catalog-backend/src/processing/TaskPipeline.ts

```diff
@@ -85,34 +89,36 @@ export function startTaskPipeline<T>(options: Options<T>) {
 
   async function pipelineLoop() {
     while (!abortSignal.aborted) {
-      if (state.inFlightCount <= lowWatermark) {
-        const loadCount = highWatermark - state.inFlightCount;
-        const loadedItems = await Promise.resolve()
-          .then(() => loadTasks(loadCount))
-          .catch(() => {
-            // Silently swallow errors and go back to sleep to try again; we
-            // delegate to the loadTasks function itself to catch errors and log
-            // if it so desires
-            return [];
-          });
-        if (loadedItems.length && !abortSignal.aborted) {
-          state.inFlightCount += loadedItems.length;
-          for (const item of loadedItems) {
-            Promise.resolve()
-              .then(() => processTask(item))
-              .catch(() => {
-                // Silently swallow errors and go back to sleep to try again; we
-                // delegate to the processTask function itself to catch errors
-                // and log if it so desires
-              })
-              .finally(() => {
-                state.inFlightCount -= 1;
-                barrier.release();
-              });
+      await withActiveSpan(tracer, 'TaskPipelineLoop', async span => {
+        if (state.inFlightCount <= lowWatermark) {
+          const loadCount = highWatermark - state.inFlightCount;
+          const loadedItems = await Promise.resolve(loadCount)
+            .then(loadTasks)
```

Comment:
I really disagree with these two changes, to me it looks absolutely bonkers to put the constant as a separately resolved argument just for the sake of it 😅 the reason for the resolve is to ensure that the work happens on a separate tick instead of in-line. Basically a setImmediate. No use mixing up the parameter passing in this. (Same below)

Reply:
I do not have a strong opinion on this one and can revert this part tomorrow, when I will have my laptop.
Took inspiration from a blog post I read some time ago https://codingwithjs.rocks/blog/the-hidden-power-of-promise-resolve

Reply:
fixed